src/main.rs

```diff
@@ -298,7 +300,7 @@ fn get_os() -> OsType {
     OsType::Other
 }
 
-fn get_languages(env_lang: Option<String>, env_language: Option<String>) -> Vec<String> {
```

Comment:
Is there any chance this could be a `Option<&str>` instead?

Reply:
Actually, both can be `Option<&str>`:

```rust
fn get_languages(env_lang: Option<&str>, env_language: Option<&str>) -> Vec<String> {
    // Language list according to
    // https://github.com/tldr-pages/tldr/blob/master/CLIENT-SPECIFICATION.md#language

    if env_lang.is_none() {
        return vec!["en".to_string()];
    }
    let env_lang = env_lang.unwrap();

    // Create an iterator that contains $LANGUAGE (':' separated list) followed by $LANG (single language)
    let locales = env_language
        .unwrap_or("")
        .split(':')
        .chain(iter::once(env_lang));

    let mut lang_list = Vec::new();
    for locale in locales {
        // Language plus country code (e.g. `en_US`)
        if locale.len() >= 5 && locale.chars().nth(2) == Some('_') {
            lang_list.push(&locale[..5]);
        }
        // Language code only (e.g. `en`)
        if locale.len() >= 2 && locale != "POSIX" {
            lang_list.push(&locale[..2]);
        }
    }

    lang_list.push("en");
    lang_list.clear_duplicates();

    lang_list.into_iter().map(str::to_string).collect()
}
```

Sorry for dictating so much, I just have to point out that old mistake (that I contributed 😄)

Although the tests still work, I think all of the `.into()` calls can be removed now. Also, there could be a `fn get_languages_from_env() -> Vec<String>` that handles the env var retrieval, so that this logic is moved out of `main` and shortened. What do you think? (feel free to leave that to someone else, if you just want to fix the clippy warnings 😄)

Reply:
I changed the function as you suggested and removed all the `.into()`s. Please have a look at the new `get_languages_from_env` function. Does it match to what you imagined?

Reply:
Oh, I didn't see your comment here 😆 Yes, almost. I started a new review below, but it's only small details. Thank you for contributing!